lib/api/core/models/repositories/profileRepository.js

```diff
@@ -186,7 +203,21 @@ module.exports = function (kuzzle) {
       }
     });
 
-    result.roles = extractRoleIds(profile.roles);
+    result.roles = [];
+    async.each(profile.roles, (role, callback) => {
```

Comment:
I may be wrong, but I'm under the impression that  this method will return an empty `result.roles` object, since getting roles from profiles will be done asynchronously with async